proposals/p2015.md

```diff
@@ -0,0 +1,237 @@
+    -   [Non-goals](#non-goals)
+-   [Details](#details)
+    -   [Syntax](#syntax)
+    -   [Usage](#usage)
+    -   [Syntax examples and considerations](#syntax-examples-and-considerations)
+        -   [Syntax example use in common fixed-size numeric types](#syntax-example-use-in-common-fixed-size-numeric-types)
+        -   [Syntax example use in arbitrary fixed-size numeric types](#syntax-example-use-in-arbitrary-fixed-size-numeric-types)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [C++ LP64 convention](#c-lp64-convention)
+    -   [Type name with length suffix](#type-name-with-length-suffix)
+    -   [Uppercase suffixes](#uppercase-suffixes)
+
+<!-- tocstop -->
+
+## Problem
+
+We want to establish a syntax for fixed-size scalar number types. These types
+include the two's complement signed integer, the unsigned integer, and the
+floating-point number.
+
+As these types are pervasive throughout the language, our goal here is to align
+on a terse, convenient, yet understandable, and ergonomic syntax to the author.
+
+## Background
+
+For developer convenience, names are given to number types that map to native
+machine register widths. These sizes typically include 8-bit, 16-bit, 32-bit,
+64-bit, and, more recently, 128-bit widths.
+
+For example, in [C++11+](https://en.cppreference.com/w/cpp/types/integer),
+integer types such as `int8_t` (8-bit two's complement signed integer type) and
+`uint16_t` (16-bit unsigned integer type) exist, among similar types for 32- and
+64-bit values. Correspondingly, you have the `i8` and `u16`
+([among others](https://doc.rust-lang.org/book/ch03-02-data-types.html#scalar-types))
+scalar integer types in Rust. And in Swift, the `Int8` and `UInt16`
+([among others](https://developer.apple.com/documentation/swift/uint8)) integer
+value types.
+
+In each case, the intent is to provide a clear and pragmatic syntax.
+
+Additional discussion around this proposal's background can be found in
+[#543](https://github.com/carbon-language/carbon-lang/issues/543).
+
+## Proposal
+
+We introduce a simple keyword-like syntax of `iN`, `uN`, and `fN` for two's
+complement integers, unsigned integers, and floating-point numbers,
+respectively. Where `N` can map to common power-of-two sizes (for example,
+`N = 8, 16, 32`) or any other fixed-width value (for example, `N = 42`). This
+structure follows the successful precedent set by Rust and LLVM development
+communities and potentially saves 40% or more on characters required compared to
+other options such as `IntN` (for example, `i16` versus `Int16`).
+
+### Non-goals
+
+-   This does not address any considerations around the `bool` type
+-   This does not provide a formal plan for the shape or mapping of the
+    underlying types
+    ([#767 comments](https://github.com/carbon-language/carbon-lang/issues/767#issuecomment-1214153375))
+-   This does not prescribe an official grammar for parsing these types
```

Comment:
Also add a non-goal saying this proposal does not address other bit sizes, such as used in a bit field.